src/test/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsTest.java

```diff
@@ -0,0 +1,248 @@
+// Copyright 2020 Google LLC
+
+package com.google.gcp_variant_transforms.library;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.api.services.bigquery.model.TableRow;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.GenotypesContext;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFormatHeaderLine;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLineCount;
+import htsjdk.variant.vcf.VCFHeaderLineType;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Units tests for VariantToBqUtils.java
+ */
+public class VariantToBqUtilsTest {
+  private static final String ALTERNATE_BASES_ALT = "alt";
+  private static final String CALLS_NAME = "name";
+  private static final String CALLS_GENOTYPE = "genotype";
+  private static final String CALLS_PHASESET = "phaseset";
+  private static final String DEFAULT_PHASESET = "*";
+  private static final String MISSING_FIELD_VALUE = ".";
+
+  VariantContext variantContext;
+  GenotypesContext genotypesContext;
+  VCFHeader vcfHeader;
+
+  @Before
+  public void constructMockClasses() {
+    variantContext = mock(VariantContext.class);
+    vcfHeader = mock(VCFHeader.class);
+    genotypesContext = mock(GenotypesContext.class);
+  }
+
+  @Test
+  public void testConvertStringValueToRightValueType_whenComparingElement_thenTrue() {
+    // test list of values
+    String integerListStr = "23,27";
+    assertThat(VariantToBqUtils.convertToDefinedType(integerListStr, VCFHeaderLineType.Integer))
+            .isEqualTo(Arrays.asList(23, 27));
+    String floatListStr = "0.333,0.667";
+    assertThat(VariantToBqUtils.convertToDefinedType(floatListStr, VCFHeaderLineType.Float))
+            .isEqualTo(Arrays.asList(0.333, 0.667));
+
+    // test int values
+    String integerStr = "23";
+    assertThat(VariantToBqUtils.convertToDefinedType(integerStr, VCFHeaderLineType.Integer))
+            .isEqualTo(23);
+
+    // test float values
+    String floatStr = "0.333";
+    assertThat(VariantToBqUtils.convertToDefinedType(floatStr, VCFHeaderLineType.Float))
+            .isEqualTo(0.333);
+
+    // test String values
+    String str = "T";
+    assertThat(VariantToBqUtils.convertToDefinedType(str, VCFHeaderLineType.String))
+            .isEqualTo("T");
+  }
+
+  @Test
+  public void testBuildReferenceBase_whenComparingElement_thenTrue() {
+    Allele refAllele = mock(Allele.class);
+    when(refAllele.getDisplayString()).thenReturn("A");   // mock reference value
```

Comment:
should we save "A" as an expectedString variable to eliminate human error between setting it as the return and checking isEqualTo? Or maybe this is okay since we're not passing in "A" to any function calls being tested--not sure.